File: src/main/java/com/prmncr/normativecontrol/services/DocumentQueue.java
package com.prmncr.normativecontrol.services;

import com.prmncr.normativecontrol.dtos.Document;
import org.springframework.lang.Nullable;
import org.springframework.stereotype.Service;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@Service
public class DocumentQueue {
    private final Map<String, Document> documentMap = new ConcurrentHashMap<>();

    public void put(Document document) {
        documentMap.put(document.getId(), document);
    }

    @Nullable
    public Document getById(String id) {
        try {
            return documentMap.get(id);
        } catch (NullPointerException e) {
            return null;
        }

    }

    public void remove(String id) {
        documentMap.remove(id);
    }
}

File: src/main/kotlin/com/prmncr/normativecontrol/services/DocumentQueue.kt
package com.prmncr.normativecontrol.services

import com.prmncr.normativecontrol.dtos.Document
import org.springframework.lang.Nullable
import org.springframework.stereotype.Service
import java.util.concurrent.ConcurrentHashMap

@Service
class DocumentQueue {
    private val documentMap: MutableMap<String, Document> = ConcurrentHashMap()
    fun put(document: Document) {
        documentMap[document.id] = document
    }

    @Nullable
    fun getById(id: String): Document? {
        return try {
            documentMap[id]
        } catch (e: NullPointerException) {
            null
        }
    }

    fun remove(id: String) {
        documentMap.remove(id)
    }
}